ring/replication_set.go

```diff
@@ -95,6 +95,122 @@ func (r ReplicationSet) Do(ctx context.Context, delay time.Duration, f func(cont
 	return results, nil
 }
 
+// DoUntilQuorum runs function f in parallel for all replicas in r.
+//
+// If r.MaxUnavailableZones is greater than zero:
+//   - DoUntilQuorum returns an error if calls to f for instances in r.MaxUnavailableZones zones return errors
+//   - Otherwise, DoUntilQuorum returns all results from all replicas in the first zones for which f succeeds
+//     for every instance in that zone (eg. if there are 3 zones and r.MaxUnavailableZones is 1, DoUntilQuorum will
+//     return the results from all instances in 2 zones, even if all calls to f succeed).
+//
+// Otherwise:
+//   - DoUntilQuorum returns an error if r.MaxErrors calls to f return errors
+//   - Otherwise, DoUntilQuorum returns all results from the first len(r.Instances) - r.MaxErrors instances
+//     (eg. if there are 6 replicas and r.MaxErrors is 2, DoUntilQuorum will return the results from the first 4
+//     successful calls to f, even if all 6 calls to f succeed).
+//
+// Any results from successful calls to f that are not returned by DoUntilQuorum will be passed to cleanupFunc,
```

Comment:
maybe mention what successful means - "returns a nil error"

Reply:
Resolved in 76c1271928b30d1e589afa09a7edd835feeba65f